### DISCUSSION TITLE
useFieldArray `field` values don't update unless triggered by `append` or `remove`

### DISCUSSION THREAD

**Discussion by iKrushYou** [react-hook-form/react-hook-form#6123] [Q&A]
As the title says, not sure if this is intended or a bug. Here's a CSB: https://codesandbox.io/s/rhf-usefieldarray-field-values-not-updating-du230

I would expect that the values in the `field` object update in realtime so that I can access them (as I would with `watch`)

If this is not intended I suppose the workaround would be to use watch on that value?

**Comment by bluebill1049** [comment]
`field` object only returns the `defaultValue`.

**Reply by iKrushYou** [reply]
Will this be done away with when you remove the `defaultValue` situation?

**Reply by bluebill1049** [reply]
no, same thing. fields will always made for defaultValue
